src/Microsoft.Health.Fhir.CosmosDb/Features/Health/FhirCosmosHealthCheck.cs

```diff
@@ -42,39 +32,16 @@ public class FhirCosmosHealthCheck : IHealthCheck
             IOptionsSnapshot<CosmosCollectionConfiguration> namedCosmosCollectionConfigurationAccessor,
             IDocumentClientTestProvider testProvider,
             ILogger<FhirCosmosHealthCheck> logger)
+            : base(
+                  documentClient,
+                  configuration,
+                  namedCosmosCollectionConfigurationAccessor,
+                  testProvider,
+                  logger)
         {
-            EnsureArg.IsNotNull(documentClient, nameof(documentClient));
-            EnsureArg.IsNotNull(configuration, nameof(configuration));
-            EnsureArg.IsNotNull(namedCosmosCollectionConfigurationAccessor, nameof(namedCosmosCollectionConfigurationAccessor));
-            EnsureArg.IsNotNull(testProvider, nameof(testProvider));
-            EnsureArg.IsNotNull(logger, nameof(logger));
-
-            _documentClient = documentClient;
-
-            EnsureArg.IsNotNull(_documentClient, optsFn: options => options.WithMessage("Factory returned null."));
-
-            _configuration = configuration;
-            _collectionConfiguration = namedCosmosCollectionConfigurationAccessor.Get(Constants.CollectionConfigurationName);
-            _testProvider = testProvider;
-            _logger = logger;
+            CollectionConfiguration = namedCosmosCollectionConfigurationAccessor.Get(Constants.CollectionConfigurationName);
         }
 
-        public async Task<HealthCheckResult> CheckAsync(CancellationToken cancellationToken = default(CancellationToken))
-        {
-            try
-            {
-                // Make a non-invasive query to make sure we can reach the data store.
-
-                await _testProvider.PerformTest(_documentClient.Value, _configuration, _collectionConfiguration);
-
-                return HealthCheckResult.Healthy("Successfully connected to the data store.");
-            }
-            catch (Exception ex)
-            {
-                _logger.LogWarning(ex, "Failed to connect to the data store.");
-
-                return HealthCheckResult.Unhealthy("Failed to connect to the data store.");
-            }
-        }
+        public override CosmosCollectionConfiguration CollectionConfiguration { get; }
```

Comment:
The return value is the same. Instead of overriding this, you could simply pass the `CollectionConfigurationName` to the base class instead?